libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -135,6 +130,80 @@ internal object MapboxNavigationTelemetry :
     private var originalRoute: DirectionsRoute? = null
     private var needStartSession = false
 
+    private val routesObserver = RoutesObserver { routes ->
+        log("onRoutesChanged. size = ${routes.size}")
+        routes.getOrNull(0)?.let {
+            if (sessionState == ACTIVE_GUIDANCE) {
+                if (originalRoute != null) {
+                    if (needHandleReroute) {
+                        needHandleReroute = false
+                        handleReroute(it)
+                    } else {
+                        log("handle ExternalRoute")
+                        sessionStop()
+                        originalRoute = it
+                        needStartSession = true
+                        startSessionIfNeedAndCan()
```

Comment:
I just noticed this while working on billing but didn't have a chance to verify yet - is this called when route refresh happens? Doesn't that interfere with the telemetry data?

Reply:
Well, this ultimately unrelated to the changes here - I can cut a ticket to investigate :upside_down_face:

Reply:
> Doesn't that interfere with the telemetry data? 

At first glance - yes, we should investigate it. nice shot @LukasPaczos 👍 
you can mention me at the a ticket, because I was working with route-refresh